### ISSUE TITLE
Possible usability issue with OAuth: cant't add app password

### ISSUE THREAD

**Issue by thommierother** [owncloud/client#7181]
Maybe I am using the wrong steps to reproduce this but here is my observation:

environment:
Server: 10.2 RC1 owncloud server, oauth app enabled
Sync client: sync client 2.5.4 build 515 (Linux)

I want to add a sync account with a permanent app password INSTEAD of going through the oauth browser auth steps upon every start of the sync application.

1. create app password in the oc account
2. add new account in the client
3. enter server address
4. client redirects to oauth authorization screen
5. oauth login successful, BUT:

My problem is that there is no option to add the app password INSTEAD on the client side. Not in the client application itself and not on the oath consent screen presented by the server. Or am I totally wrong?

two possible options:
a) entry  in the sync client to add app password application AFTER doing the Oath process one time
b) additional button in the Oauth consent screen to do this DURING the first Oauth process

**Comment by michaelstingl** [comment]
> My problem is that there is no option to add the app password INSTEAD on the client side.

@thommierother Yes, this is exactly the way it was designed. But you only need to execute the OAuth 2.0 once. OAuth 2.0 login should never expire. If you need to perform the steps again and again, you encountered a bug, and we should have a closer look. Could you elaborate on your desktop setup. Maybe open a new issue?

> I think of a use case where a user has oauth active for web access and uses the sync client in parallel.

OAuth 2.0 is only used for client/app authorisation, has nothing to do with the login to the web UI. This use case makes no sense.

**Comment by thommierother** [comment]
@michaelstingl 
I see the OAuth process appearing again and again upon every new start of the client, so it looks like a bug. Give me some hints for deeper debugging .. . The client is 2.5.4 build 515 for OpenSUSE Leap 15

**Comment by michaelstingl** [comment]
@thommierother You could capture client log (start with `--logfile` parameter) and server log (`owncloud.log`). Here you can find more information:
https://doc.owncloud.com/desktop/2.5/troubleshooting.html#log-files

Maybe create a [secret Gist](https://gist.github.com/) and send the URL to `desktop-<at>-owncloud.com`

**Comment by thommierother** [comment]
@michaelstingl: I re-tested this with the RC3 server code and the same client version (2.5.4) and now the behaviour has changed: If I add the app password upon creation of the client account through the Web UI, there is no re-appearing oauth authentication process any more, even not after a client restart. So either something changed between RC1 and RC3 or it has to do with the update of oauth app from 0.4.0 to 0.4.1 (although the only change was https://github.com/owncloud/oauth2/pull/197). Closing this issue now.
